File: PrimeiraEtapa/Semana4/SliderRGB/SliderRGB/ViewController.m
//
//  ViewController.m
//  SliderRGB
//
//  Created by bepid on 04/11/14.
//  Copyright (c) 2014 ifce. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.r = 1.0;
    self.b = 1.0;
    self.g = 1.0;
    //[self mudaCor];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)SliderRed:(UISlider *)sender {
    self.r = sender.value;
    _redField.backgroundColor = [UIColor colorWithRed:self.r green:0.0 blue:0.0 alpha:1.0];
    _redField.text = [NSString stringWithFormat:@"%2.f", [sender value]*255 ] ;
    [self mudaCor];
}

- (IBAction)SliderGreen:(UISlider *)sender {
    self.g = sender.value;
    _greenField.backgroundColor = [UIColor colorWithRed:0.0 green:self.g blue:0.0 alpha:1.0];
    _greenField.text = [NSString stringWithFormat:@"%2.f", [sender value]*255 ] ;
    [self mudaCor];
}


- (IBAction)SliderBlue:(UISlider *)sender {
    self.b = sender.value;
    _blueField.backgroundColor = [UIColor colorWithRed:0.0 green:0.0 blue:self.b alpha:1.0];
    _blueField.text = [NSString stringWithFormat:@"%2.f", [sender value]*255 ] ;
    [self mudaCor];
}



-(void) mudaCor
{
    self.view.BackgroundColor =[UIColor colorWithRed:self.r green:self.g blue:self.b alpha:1.0];
}


@end

File: Quinta Etapa/AVFoundationTest/AVFoundation-Test/ViewController.swift
//
//  ViewController.swift
//  AVFoundation-Test
//
//  Created by Diego Vidal on 18/08/15.
//  Copyright (c) 2015 Diego Vidal. All rights reserved.
//

import UIKit
import AVFoundation
import AVKit

class ViewController: UIViewController {

    var playerController = AVPlayerViewController()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func actionButton(sender: UIButton) {
        
        if let url = NSBundle.mainBundle().URLForResource("IMG_0255", withExtension: ".MOV"){
            let playerItem = AVPlayerItem(URL: url)
            let player = AVPlayer(playerItem: playerItem)
            playerController.player = player
            
            self.presentViewController(playerController, animated: true, completion: nil)
            
            player.play()
            
            var button = UIButton.buttonWithType(UIButtonType.System) as! UIButton
            button.frame = CGRectMake(100, 100, 100, 50)
            button.addTarget(self, action: "buttonAction", forControlEvents: UIControlEvents.TouchUpInside)
           button.backgroundColor = UIColor.blueColor()
            // não pode interagir
            //playerController.contentOverlayView
            
            playerController.view.addSubview(button)
        }
    }
    
    func buttonAction(){
        playerController.player.rate = playerController.player.rate*2
    }

}

